simulatingrisk/hawkdovemulti/batch_run.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python
+
+import argparse
+import csv
+from datetime import datetime
+import multiprocessing
+import os
+
+from tqdm.auto import tqdm
+
+from mesa.batchrunner import _make_model_kwargs, _collect_data
+
+from simulatingrisk.hawkdovemulti.model import HawkDoveMultipleRiskModel
+
+
+neighborhood_sizes = list(HawkDoveMultipleRiskModel.neighborhood_sizes)
+
+# combination of parameters we want to run
+params = {
+    "grid_size": [10, 25, 50],  # 100],
+    "risk_adjustment": ["adopt", "average"],
+    "play_neighborhood": neighborhood_sizes,
+    "observed_neighborhood": neighborhood_sizes,
+    "adjust_neighborhood": neighborhood_sizes,
+    "hawk_odds": [0.5, 0.25, 0.75],
+    "adjust_every": [2, 10, 20],
+    "risk_distribution": HawkDoveMultipleRiskModel.risk_distribution_options,
+    "adjust_payoff": HawkDoveMultipleRiskModel.supported_adjust_payoffs,
+    # random?
+}
+
+
+# method for multiproc running model with a set of params
+def run_hawkdovemulti_model(args):
+    run_id, iteration, params, max_steps = args
+    # simplified model runner adapted from mesa batch run code
+
+    model = HawkDoveMultipleRiskModel(**params)
+    while model.running and model.schedule.steps <= max_steps:
+        model.step()
+
+    # collect data for the last step
+    step = model.schedule.steps - 1
+
+    model_data, all_agents_data = _collect_data(model, step)
+
+    # combine run id, step, and params, with collected model data
+    run_data = {"RunId": run_id, "iteration": iteration, "Step": step}
+    run_data.update(params)
+    run_data.update(model_data)
+
+    agent_data = [
+        {
+            "RunId": run_id,
+            "iteration": iteration,
+            "Step": step,
+            **agent_data,
+        }
+        for agent_data in all_agents_data
+    ]
+
+    return run_data, agent_data
+
+
+def batch_run(
+    params, iterations, number_processes, max_steps, progressbar, file_prefix=""
+):
```

Comment:
Should the argument `file_prefix` have a default value? Or, should this default value be specified in `main` (default for `argparse.file_prefix`)?

Reply:
that's true, it would probably make more sense to set the default in the argsparse config the way this is currently written. It would certainly be more consistent with how I'm handling the other options.